main.py

```diff
@@ -670,6 +670,41 @@ def delete_link_metadata(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         return JSONResponse("Operation successful")
 
+    @rest('v3/links/{link_id}', methods=['DELETE'])
+    def delete_link(self, request: Request) -> JSONResponse:
+        """Delete a disabled link from topology.
+         It won't work for link with other statuses.
+        """
+        link_id = request.path_params["link_id"]
+        try:
+            with self._links_lock:
+                link = self.links[link_id]
+                if link.status != EntityStatus.DISABLED:
+                    raise HTTPException(409, detail="Link is not disabled.")
+                if link.endpoint_a.link and link == link.endpoint_a.link:
+                    switch = link.endpoint_a.switch
+                    link.endpoint_a.link = None
+                    link.endpoint_a.nni = False
+                    self.topo_controller.upsert_switch(
+                        switch.id, switch.as_dict()
+                    )
+                if link.endpoint_a.link and link == link.endpoint_b.link:
```

Comment:
@Alopalao, I was about to approve this PR, just noticed `link.endpoint_a.link` here looks off. Did you mean `link.endpoint_b.link`?

Let's also make sure this branch is unit tested, currently it's not covered.

Reply:
Fixed, also with it, it is also covered now.